frontend/src/__tests__/cypress/cypress/pages/pipelines/archiveRunModal.ts

```diff
@@ -0,0 +1,25 @@
+import { Modal } from '~/__tests__/cypress/cypress/pages/components/Modal';
+
+class ArchiveRunModal extends Modal {
+  protected testId;
+
+  constructor(testId = 'archive-run-modal') {
+    super('Archive run?');
+    this.testId = testId;
+  }
+
+  find() {
+    return cy.findByTestId(this.testId).parents('div[role="dialog"]');
+  }
+
+  findConfirmInput() {
+    return cy.findByTestId('confirm-archive-input');
```

Comment:
```suggestion
    return this.find().findByTestId('confirm-archive-input');
```

should it be finding within the modal?

Reply:
Yes, it should, I'll update.